crates/shared/src/price_estimation/factory.rs

```diff
@@ -189,25 +189,35 @@ impl<'a> PriceEstimatorFactory<'a> {
                 let estimator = self.get_estimator(driver)?.native.clone();
                 Ok((
                     driver.name.clone(),
-                    Arc::new(NativePriceEstimator::new(
-                        Arc::new(self.sanitized(estimator)),
-                        self.network.native_token,
-                        native_token_price_estimation_amount,
+                    Arc::new(InstrumentedPriceEstimator::new(
+                        NativePriceEstimator::new(
+                            Arc::new(self.sanitized(estimator)),
+                            self.network.native_token,
+                            native_token_price_estimation_amount,
+                        ),
+                        driver.name.to_string(),
+                    )),
+                ))
+            }
+            NativePriceEstimatorSource::OneInchSpotPriceApi => {
+                let name = "OneInchSpotPriceApi".to_string();
```

Comment:
UpperCamelCase should be avoided since `driver.name` uses flat case. We should use a similar format for other names as well. If not flatcase everywhere, maybe snake_case.